From 50e8861294bf4b8aa624c575b5d4c1810714327a Mon Sep 17 00:00:00 2001
From: root <root@hp-dl580g8-05.rhts.eng.pek2.redhat.com>
Date: Fri, 1 Jun 2018 09:49:35 -0400
Subject: [PATCH] memsep: use _exit() to exit child

Avoid glibc atexit handlers to update file offset on
/proc/$pid/maps, because it confuses parent, which can
get stuck re-reading same content.

Signed-off-by: Jan Stancek <jstancek@redhat.com>
---
 src/memsep.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/memsep.c b/src/memsep.c
index fd1cf2d..175cef4 100644
--- a/src/memsep.c
+++ b/src/memsep.c
@@ -84,7 +84,7 @@ void sig_handler(int sig)
 	if (debug) {
 		fprintf(stderr, "caught the fault %d\n", sig);
 	}
-	exit(0);
+	_exit(0);
 }
 
 /************************************************************************/
@@ -145,7 +145,7 @@ void write_read_mem(int *ptr, int write_flag)
 			}
 			*ptr = rand();
 		}
-		exit(-1);
+		_exit(-1);
 	}
 	else if (pid == -1) {
 		// Error condition 
-- 
2.17.0

